src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,312 @@
+            if (arithcalc and not bitcalc and not urscalc): # If we need bitwise, arithmetic calculations
+                                                            # or unsigned right shift
+                calc = 1
+            elif (bitcalc and not arithcalc and not urscalc):
+                calc = 2
+            elif (urscalc and not arithcalc and not bitcalc):
+                calc = 3
+            elif (arithcalc and bitcalc and not urscalc):
+                calc = random.randint(1, 2) # 1 or 2
+            elif (bitcalc and urscalc and not arithcalc):
+                calc = random.randint(2, 3) # 2 or 3
+            elif (arithcalc and urscalc and not bitcalc):
+                calc = (1 + random.randint(0, 1) * 2) # Must be 1 or 3
+            elif (arithcalc and bitcalc and urscalc):
+                calc = random.randint(1, 3) # 1, 2 or 3
+
+            gs = GenSequence(options, calc)
+            if (calc == 1): # Setting maximum depending on operation type
+                MINNUM = _constants.min
+                MAXNUM = _constants.max
+            elif (calc == 2):
+                MINNUM = _constants.bitmin
+                MAXNUM = _constants.bitmax
+            else:
+                MINNUM = _constants.uint32min
+                MAXNUM = _constants.uint32max
+            last = gs.operands[0]
+            # Converting expression, expected value, false value, and function call to string format
+            expression = ""
+            expression += ("%s%d" % ("(" * (gs.num_operands - 1), last))
+            for i in range(options.operand_count - 1):
+                result = gs.operators_list[gs.operators[i]](last, gs.operands[i + 1])
+                expression += (" %s %d)" % (result["oper"], result["op2"]))
+                if (result["oper"] == "/"):
+                    expression += " | 0)"
+                    expression = "(" + expression
+                gs.operands[i + 1] = result["op2"]
+                last = result["sum"]
+            expression += "\n"
+            expected_value = last
+            # Random number wich must not be equal with the expected value
+            if((random.randint(0, 1) and last != MAXNUM) or last == MINNUM ):
+                N1 = random.randint(math.ceil(last + 1), MAXNUM)
+                N2 = random.randint(math.ceil(last + 1), MAXNUM)
+                N3 = random.randint(math.ceil(last + 1), MAXNUM)
+            else:
+                N1 = random.randint(MINNUM, math.floor(last - 1))
+                N2 = random.randint(MINNUM, math.floor(last - 1))
+                N3 = random.randint(MINNUM, math.floor(last - 1))
+            false_values = "[(%s), (%s), (%s)]" % (str(N1), str(N2), str(N3))
+            file_content += ("validate_numeric((%s), (%s), (%s));\n" % (expression, str(expected_value), false_values))
+        write_file._write_file(options.output,  _GENERATED_FILENAME.format(NUMBER = file_count), file_content)
+
+def main(options):
```

Comment:
I still do not like the this kind of main functions. It is so simple that you could rename the `generate_binary_tests` to `main` instead! Or if you refactor the function as I said above, it will have some valuable actions here as well.